src/services/api.js

```diff
@@ -0,0 +1,154 @@
+          }
+        });
+      }
+
+      /** replace response to plain format */
+      if (responses.responses) {
+        Object.keys(responses.responses).forEach(item => {
+          for (let response of responses.responses[item]) {
+            if (response.value && response.value.src && response.value.ptr !== undefined) {
+              response.value = responses.dataSources[response.value.src][response.value.ptr];
+            }
+          }
+
+          responses.responses[item] = responses.responses[item].filter((response) => response.value);
+          if (responses.responses[item].length === 0) {
+            delete responses.responses[item];
+          }
+        })
+      }
+
+      if (responses.items) {
+        for (let itemId in responses.items) {
+          const item = responses.items[itemId];
+          responses.items[itemId] = {
+            ...itemAttachExtras(itemTransformJson(item), itemId),
+            original: item.original,
+            activityId: item.activityId
+          }
+        }
+      }
+
+      if (responses.itemReferences) {
+        for (let version in responses.itemReferences) {
+          for (let itemId in responses.itemReferences[version]) {
+            const id = responses.itemReferences[version][itemId];
+            if (id) {
+              const item = responses.items[id];
+              responses.itemReferences[version][itemId] = item;
+            }
+          }
+        }
+      }
+
+      if (responses.activities) {
+        for (let activityId in responses.activities) {
+          const activity = responses.activities[activityId];
+          if (activity[ORDER]) {
+              delete activity[ORDER];
+          }
+
+          responses.activities[activityId] = {
+            ...activityTransformJson(activity, []),
+            original: activity.original
+          }
+        }
+      }
+
+      return { ...responses, appletId };
+    });
+  });
+  return Promise.all(requests)
+    .then(transformResponses);
+};
+
+const uploadFiles = (authToken, response, item) => {
+  const answers = R.pathOr([], ['responses'], response);
+
+  // Each "response" has number of "answers", each of which may have a file
+  // associated with it
+  const uploadRequests = Object.keys(answers).reduce((accumulator, key) => {
+    const answer = answers[key];
+
+    // Surveys with a "uri" value and canvas with a "uri" will have files to upload
+    let file;
+    if (R.path(['survey', 'uri'], answer)) {
+      file = { uri: answer.survey.uri, filename: answer.survey.filename, type: 'application/octet' };
+    } else if (R.path(['canvas', 'uri'], answer)) {
+      file = { uri: answer.canvas.uri, filename: answer.canvas.filename, type: 'application/jpg' };
+    } else if (answer && answer.uri && answer.filename) {
+      file = { uri: answer.uri, filename: answer.filename, type: 'application/octet' };
+    } else {
+      return accumulator; // Break early
+    }
```

Comment:
@nikhilrayaprolu Good to have `file` variable like this:
let file = {}
if () {
> file = {
>     …
>     …
> }

}